### ISSUE TITLE
Don't treat comments starting with `noqa` as a `noqa` comment if they're clearly not

### ISSUE THREAD

**Issue by jakkdl** [astral-sh/ruff#7333]
**example file**
```python
# noqa is used extensively in this file [...]
```
**output**
```
$ ruff --select ALL foo.py
...
warning: Invalid `# noqa` directive on foo.py:1: expected `:` followed by a comma-separated list of codes (e.g., `# noqa: F401, F841`).
...
```
(using `--isolated` removes the warning)

No settings, version `0.0.289`.

**Real life incident**: https://github.com/python-trio/trio/pull/2795/files/78d67c11d99d64c19f0e5182608b6653ad32924b#r1323337111

**Minimal suggested fix**: Ignore warning if the line is blank, and the comment has trailing characters after `noqa[^:]`.

**Comment by charliermarsh** [comment]
I think my preference is to retain our current behavior. I know it's annoying in certain cases, but it's possible that the user wrote a line that starts with `# noqa` followed by an explanation and expected it to work as a blanket ignore comment.
